nni/retiarii/execution/base.py

```diff
@@ -93,18 +95,21 @@ def _final_metric_callback(self, trial_id: int, metrics: MetricData) -> None:
         for listener in self._listeners:
             listener.on_metric(model, metrics)
 
-    def query_available_resource(self) -> List[WorkerInfo]:
-        raise NotImplementedError  # move the method from listener to here?
+    def query_available_resource(self) -> int:
+        return self.resources
 
     @classmethod
     def trial_execute_graph(cls) -> None:
         """
         Initialize the model, hand it over to trainer.
         """
         graph_data = BaseGraphData.load(receive_trial_parameters())
-        with open('_generated_model.py', 'w') as f:
+        random_str = ''.join(random.choice(string.ascii_uppercase + string.digits) for _ in range(6))
```

Comment:
We may check if the random_str is repeated, although the probability is low.
A lock is needed if trial_execute_graph can be executed concurrently.

Reply:
yes, we can check that.
why a lock is required?

Reply:
We may need a set to check the existed model names. If two models are generated concurrently in two threads and they generate the same random_str (although with a very very low probability), the random_str is not in the set when checking its existence. Then both models will have the same random_str.

Reply:
Another way out is to decrease the probability. With a sufficiently long hexdigit (e.g., 16 characters), this is almost impossible.

Reply:
good suggestions, i will fix this one in the next pr